src/Features/ERDDAP/hooks/BuoyBarnComponents.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Components to use React-Query in a common way
+ */
+import * as React from "react"
+import { Alert } from "reactstrap"
+
+import { PlatformFeature, PlatformFeatureCollection } from "../types"
+
+import { usePlatforms } from "./buoyBarn"
+
+interface BaseProps {
+  /** Override default loading alert */
+  loading?: JSX.Element
+  /** Override default error alert */
+  error?: JSX.Element
+}
+
+interface UsePlatformsProps extends BaseProps {
+  /** Pass platforms to child elements */
+  children: (props: UsePlatformsRenderProps) => JSX.Element
```

Comment:
Codacy found an issue: ['JSX' is not defined.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)